calyx/src/analysis/read_write_set.rs

```diff
@@ -34,6 +34,49 @@ impl ReadWriteSet {
             .collect()
     }
 
+    /// returns the register cells whose out port is read anywhere in the given
+    /// assignments
+    pub fn register_reads(assigns: &[ir::Assignment]) -> Vec<RRC<ir::Cell>> {
+        let guard_ports = assigns.iter().flat_map(|assign| {
+            assign.guard.all_ports().into_iter().filter_map(|port_ref| {
+                let port = port_ref.borrow();
+                if let ir::PortParent::Cell(cell_wref) = &port.parent {
+                    if &port.name == "out" {
+                        Some(Rc::clone(&cell_wref.upgrade()))
+                    } else {
```

Comment:
Instead of multiple `else { None }` cases, you can use `return Some(...)` in the branch and then the fall through case can be `None`.